src/app/home/components/DataList/Plans/MigrationsTable.tsx

```diff
@@ -80,42 +85,45 @@ export default class MigrationsTable extends React.Component<any, any> {
         .toString();
       status.end = endTime ? moment(endTime).format('LLL') : 'TBD';
 
-      const migPhase = migration.status.phase;
       const serverErrorMessage = migration.status.errors;
       if (serverErrorMessage) {
-        status.phase = serverErrorMessage.pop();
+        status.stepName = serverErrorMessage.pop();
         status.progress = null;
         return status;
       } else {
-        switch (migPhase) {
-          case 'WaitOnResticRestart':
-            status.phase = 'Waiting';
-            status.progress = 10;
-            break;
-          case 'BackupStarted':
-            status.phase = 'Backup started';
-            status.progress = 40;
-            break;
 
-          case 'WaitOnBackupReplication':
-            status.phase = 'Waiting';
-            status.progress = 50;
-            break;
-          case 'RestoreStarted':
-            status.phase = 'Restoring....';
-            status.progress = 60;
-            break;
-          case 'Completed':
-            status.phase = 'Succeeded';
-            status.progress = null;
-            break;
-          case 'BackupFailed':
-            status.phase = 'Backup Failed';
-            status.progress = null;
-            break;
-          default:
-            status.phase = migPhase;
-            break;
+        if (migration.status.conditions.length) {
```

Comment:
We have accumulated some tech debt here. Going forward I would like to propose that we move this type of logic out of the presentation layer & into selectors. This has a few benefits: It is much easier to test (if we ever write tests) and we can leverage reselect to memoize results which allows for fewer re-renders. 

For example, we could create a selector function that takes the current migration state as a param, calculate the migrations status within the function body, & return the derived status for use within the UI.

To do this we would need to pinpoint where the ui state is being updated from the api polling request. From there, instead of passing in the state directly from redux to the UI component, we would pass the redux state-> selector function -> ui component. 

here is an example of what has been done at the plan status level
https://github.com/fusor/mig-ui/blob/787f935b381828a135b340d66d0bdfed006536cf/src/app/plan/duck/selectors.ts#L10